indexing-service/src/main/java/org/apache/druid/indexing/overlord/supervisor/SupervisorResource.java

```diff
@@ -395,6 +395,25 @@ public Response shutdown(@PathParam("id") final String id)
     return terminate(id);
   }
 
+  @POST
+  @Path("/{id}/{taskGroupId}/stopEarly")
```

Comment:
```suggestion
  @Path("/{id}/taskGroups/restart")
```

I think we should add the keyword `taskGroups` to the API path to indicate this is operating on the taskGroups of the supervisor. Also, it would be nice if the API accepted a list of taskGroups instead of a single taskGroup so that users can call it with many taskGroups. Something like

```
PUT .../supervisor/{id}/taskGroups/restart

{ "taskGroupIds": [...]}
```